src/esp/sim/BatchReplayRenderer.cpp

```diff
@@ -106,6 +108,49 @@ BatchReplayRenderer::BatchReplayRenderer(
           Mn::Matrix4::from(rotation.toMatrix(), translation);
     }
 
+    void changeLightSetup(const esp::gfx::LightSetup& lights) override {
+      if (!renderer_.maxLightCount()) {
+        ESP_WARNING() << "Attempted to change" << lights.size()
+                      << "lights for scene" << sceneId_
+                      << "but the renderer is configured without lights";
+        return;
+      }
+
+      renderer_.clearLights(sceneId_);
+      for (std::size_t i = 0; i != lights.size(); ++i) {
+        const gfx::LightInfo& light = lights[i];
+        CORRADE_INTERNAL_ASSERT(light.model == gfx::LightPositionModel::Global);
+
+        const std::size_t nodeId = renderer_.addEmptyNode(sceneId_);
+
+        /* Clang Tidy, you're stupid, why do you say that "lightId" is not
+           initialized?! I'm initializing it right in the branches below, I
+           won't zero-init it just to "prevent bugs" because an accidentally
+           zero-initialized variable is *also* a bug, you know? Plus I have
+           range asserts in all functions so your "suggestions" are completely
+           unhelpful. */
+        std::size_t lightId;  // NOLINT
+        if (light.vector.z()) {
+          renderer_.transformations(sceneId_)[nodeId] =
+              Mn::Matrix4::translation(light.vector.xyz());
+          lightId = renderer_.addLight(sceneId_, nodeId,
+                                       gfx_batch::RendererLightType::Point);
+        } else {
+          /* The matrix will be partially NaNs if the light vector is in the
```

Comment:
Well that sounds pretty shady. You're really okay with NaNs floating around here? What if one day we want to assert that all transformations are orthonormal?

In past codebases, I used to have a "getArbitraryOrthogonalVector" and/or "getArbitraryLookatTransform" for this kinda thing.

Reply:
I'm okay with that, yes. I view it as the NaNs signifying that the X/Y basis is unimportant and only Z matters, and it works as expected even in presence of matrix transformations -- NaNs stay where they should stay:

```py
>>> Matrix4.rotation_x(Deg(30.0))@\
    Matrix4.look_at(Vector3(), Vector3.y_axis(), Vector3.y_axis())
Matrix(-nan, -nan, 0, 0,
       -nan, -nan, -0.866025, 0,
       -nan, -nan, -0.5, 0,
       -nan, -nan, 0, 1)
```

If it eventually causes asserts for whatever reason, I can fix those, but even orthonormal transforms are sometimes impossible [such as in this case of meshes flattened along a certain axis](https://github.com/facebookresearch/habitat-sim/pull/1868). And inventing an always-orthogonal transformation just to avoid NaNs would only mean having extra code to test and it'd be all discarded in the end anyway.

Reply:
Some bold ideas from Mosra today, haha. Another counterpoint: what if someone tries to debug-render all the node coordinate frames as red/green/blue axes or whatever? Now you have NaNs in vertices you're submitting to the driver.

Reply:
That's also fine, it'll show the blue axis only (and yes, I'm doing exactly that regularly) ;)

Reply:
Hmm, that's right -- a mesh transformed by this matrix would be all NaNs. A visualizer using the base axes directly wouldn't tho, it would show just the blue axis, which is maybe what is meant to be visualized?

I'm taking it too far here, am I :D

Reply:
>it would show just the blue axis, which is maybe what is meant to be visualized?

Ohhhh, that's going to be a mystery rabbit hole for whatever poor junior SWE attempts this visualization.

Speaking of rabbit holes, I don't like dragging out these little code review nits, and I don't want to get into a mode of demanding changes, so I'll ask you to make the final decision here!

Reply:
I added a TODO -- once I have a convenience utility for this in Magnum, I change to it instead of `lookAt()`.